specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualMachines.json

```diff
@@ -0,0 +1,801 @@
+        },
+        "provisioningState": {
+          "type": "string"
+        },
+        "status": {
+          "type": "object",
+          "description": "MOCVirtualMachineStatus defines the observed state of MOCVirtualMachine"
+        }
+      },
+      "description": "MOCVirtualMachineSpec defines the desired state of MOCVirtualMachine"
+    },
+    "virtualmachines": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "x-ms-azure-resource": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/virtualmachinesProperties"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource Type"
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        },
+        "extendedLocation": {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "The extended location type."
+            },
+            "name": {
+              "type": "string",
+              "description": "The extended location name."
+            }
+          }
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "description": "The virtualmachines resource definition."
+    },
+    "hardwareProfileUpdate": {
+      "type": "object",
+      "properties": {
+        "vmSize": {
+          "type": "string"
+        },
+        "processors": {
+          "type": "integer",
+          "format": "int32",
+          "description": "number of processors for the virtual machine"
+        },
+        "memoryGB": {
+          "type": "integer",
+          "format": "int32",
+          "description": "RAM in gb for the virtual machine"
+        }
+      },
+      "description": "HardwareProfile - Specifies the hardware settings for the virtual machine."
+    },
+    "networkProfileUpdate": {
+      "type": "object",
+      "properties": {
+        "networkInterfaces": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "id": {
```

Comment:
>id [](http://example.com/codeflow?start=15&length=2)

Suggest resourceId for the name of this property.

Reply:
Agreed. That said, the API was published to our customers with the "id" property name (this API was copied from RPSaaSdev, where it was reviewed before). In order to avoid introducing any breaking change, we'd like to keep "id" in this version of the API and fix this in a future revision. Does this plan sound good?

I added this workitem to the pending task here: https://dev.azure.com/msazure/msk8s/_boards/board/t/ARC-HCI/Stories/?workitem=12679581